azurelinuxagent/ga/update.py

```diff
@@ -280,10 +281,15 @@ def run(self):
                             available_agent.name)
                     break
 
-                utc_start = datetime.utcnow()
 
-                last_etag = exthandlers_handler.last_etag
-                exthandlers_handler.run()
+                while True:
```

Comment:
Not a fan of this construct - we will restart the process on the next iteration anyway, not sure if there is a need for this.

Reply:
Brendan had concerns about rerunning the first part of the loop upon the goal state going stale - specifically the self update logic. I put this construct in as a stopgap until I can investigate further.